site/content/en/docs/tasks/monitor_pending_workloads/pending_workloads_on_demand.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "Pending Workloads on-demand"
+date: 2023-12-04
+weight: 3
+description: >
+#   Monitor pending workloads.
+---
+
+This page shows you how to monitor pending workloads with VisibilityOnDemand feature.
+
+The intended audience for this page are [batch administrators](/docs/tasks#batch-administrator).
+
+From version v0.6.0, Kueue provides the ability for a batch administrators to monitor
+the pipeline of pending jobs, and help users to estimate when their jobs will
+start.
+
+## Before you begin
+
+Make sure the following conditions are met:
+
+- A Kubernetes cluster is running.
+- The kubectl command-line tool has communication with your cluster.
+- [Kueue is installed](/docs/installation) in version v0.6.0 or later.
+
+## Enabling feature VisibilityOnDemand
+
+VisibilityOnDemand is an `Alpha` feature disabled by default, check the [Change the feature gates configuration](/docs/installation/#change-the-feature-gates-configuration) section of the [Installation](/docs/installation/) for details.
+
+## Monitor pending workloads on demand
+
+> _Available in Kueue v0.6.0 and later_
+
+To install a simple setup of cluster queue, run the following command:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/admin/single-clusterqueue-setup.yaml
+```
+
+To install two local queues, run the following command:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/admin/localqueues-setup.yaml
+```
+
+Now, let's create a job that will block cluster queue, so no job is admitted
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/jobs/blocking-job.yaml
+```
+
+And a few job beloning to LocalQueue `team-a-queue`
+
+```shell
+for i in {1..5}; do kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/jobs/team-a-job.yaml; done
+```
+
+and to LocalQueue `team-b-queue`
+
+```shell
+for i in {1..5}; do kubectl create -f https://raw.githubusercontent.com/kubernetes-sigs/kueue/main/site/static/examples/jobs/team-b-job.yaml; done
+```
+
+### Cluster Queue visibility
+
+To view pending workloads in ClusterQueue run the following command:
+
+```shell
+kubectl get --raw /apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads
+```
+
+You can pass optional query parameters:
+- limit `<int>` - 1000 on default. It indicates max number of pending workloads that should be fetched.
+- offset `<int>` - 0 by default. It indicates position of the first pending workload that should be fetched, starting from 0.
+
+To view only the first 3 pending workloads use:
+```shell
+kubectl get --raw /apis/visibility.kueue.x-k8s.io/v1alpha1/clusterqueues/cluster-queue/pendingworkloads\?limit\=3
```

Comment:
... \?limit\=3

Why do we use the escape character "\\\" here?

Reply:
If one doesn't use quotation marks in the command "\" are necessary. I agree that using quotation marks and omitting "\" is a cleaner way so I'll fix that